app/src/main/java/com/example/bottomnav/Components.java

```diff
@@ -5,39 +5,107 @@
 import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTNamedTypeSpecifier;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTSimpleDeclSpecifier;
 import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTSimpleDeclaration;
 
 import java.util.Optional;
 public class Components {
     public CPPASTName name;
-    public IToken typeQualifier;
+    public CPPASTName typeQualifier;
     public CPPASTName type;
     public Optional<CPPASTEqualsInitializer> init;
 
-    public Components(CPPASTName givenName, IToken givenTQ,
+    public Components(CPPASTName givenName, CPPASTName givenTQ,
                       CPPASTName givenType, Optional<CPPASTEqualsInitializer> givenInit) {
         name = givenName;
         typeQualifier = givenTQ;
         type = givenType;
         init = givenInit;
     }
 
-    public static Optional<Components> Deconstruct(CPPASTSimpleDeclaration declaration) throws Exception {
-        CPPASTNamedTypeSpecifier specifier = (CPPASTNamedTypeSpecifier) declaration.getDeclSpecifier();
-        CPPASTDeclarator declarator = (CPPASTDeclarator) declaration.getDeclarators()[0];
+    public static Optional<Components> getComponents(CPPASTSimpleDeclaration dec) throws Exception {
+        if (dec.getDeclSpecifier() instanceof CPPASTSimpleDeclSpecifier) {
+            return makeOptComp((CPPASTSimpleDeclSpecifier) dec.getDeclSpecifier(),
+                    (CPPASTDeclarator) dec.getDeclarators()[0]);
+        } else {
+            return makeOptComp((CPPASTNamedTypeSpecifier) dec.getDeclSpecifier(),
+                    (CPPASTDeclarator) dec.getDeclarators()[0]);
+        }
+    }
+
+    /**
+     *
+     * @param simpleSpec
+     * @param declarator
+     * @return Optional<Components>
+     * @throws Exception
+     *
+     * Difference: Simple specifiers don't quantifier types and do not have name object
+     */
+    private static Optional<Components> makeOptComp(CPPASTSimpleDeclSpecifier simpleSpec,
+                                                    CPPASTDeclarator declarator) throws Exception {
+        CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) declarator.getInitializer();
+        Optional<CPPASTEqualsInitializer> initOpt = makeOptInit(init);
+
+        if (declarator != null && simpleSpec != null) {
+            CPPASTName TQname = new CPPASTName();
+            TQname.setName(simpleSpec.getSyntax().getCharImage());
+            CPPASTName primName = new CPPASTName();
+            primName.setName(getPrimitiveType(simpleSpec.getType()));
+            return Optional.of(new Components((CPPASTName) declarator.getName(),
+                    TQname,
+                    primName,
+                    initOpt));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * This parser is a little goofy, instead of returning toString version of prim. type, it
+     * returns a number
+     */
+    private static char[] getPrimitiveType(int primType) {
```

Comment:
This is weird - does the stock AST parser actually map types to a number? If so - please make it clear in the comments that this is the workaround for the AST parser, instead of this parser - which implies your code (that we can fix) instead of external library code (that would be more of an uphill battle to fix)